src/Azure/Azure.Quantum.Client/Authentication/TokenFileCredential.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+namespace Microsoft.Azure.Quantum.Authentication
+{
+    using System;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using global::Azure.Core;
+    using global::Azure.Identity;
+    using Microsoft.Azure.Quantum.Utility;
+
+    /// <summary>
+    /// Implements a custom TokenCredential to use a local file as the source for an AzureQuantum token.
+    ///
+    /// It will only use the local file if the AZUREQUANTUM_TOKEN_FILE environment variable is set, and references
+    /// an existing json file that contains the access_token and expires_on timestamp in milliseconds.
+    ///
+    /// If the environment variable is not set, the file does not exist, or the token is invalid in any way(expired, for example),
+    /// then the credential will throw CredentialUnavailableError, so that DefaultQuantumCredential can fallback to other methods.
```

Comment:
```suggestion
    /// then the credential will throw <see cref="CredentialUnavailableError" />, so that DefaultQuantumCredential can fallback to other methods.
```
It may also be worth refactoring into a `<throws>`.